index.php

```diff
@@ -115,6 +118,13 @@ function failed($reason) {
 require($CFG->dirroot . '/lib/setup.php');
 require_once($CFG->libdir.'/filelib.php');
 
+// IP Locking, check for remote IP in validated list, if not, exit
+$allowedips = get_config('tool_heartbeat','ipconfig');
```

Comment:
be careful with naming things, eg here we have a setting called 'ipconfig' which isn't self explanatory, and it's not clear whether it's a whitelist or black list or something else.

You set it to a var called $allowedips so the inconsistency jumps out, I think they should have the same name in this particular case